### ISSUE TITLE
ESP8266 WemosD1 Boot failure - Suspected I2c error

### ISSUE THREAD

**Issue by PrathikGopal** [esphome/issues#1882]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

Hass.io
**ESP (ESP32/ESP8266, Board/Sonoff):**

ESP8266 - Wemos D1 mini
**ESPHome version (latest production, beta, dev branch)**

v1.16.2
**Affected component:**

ssd1306

**Description of problem:**
Boot Failure, Error on logs. This was working previously and broke only after upgrade recently

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: kbedroom
  platform: ESP8266
  board: d1_mini

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_pass
  fast_connect: true
  manual_ip:
    static_ip: !secret kbedroomsensor_ip
    gateway: !secret ip_gateway
    subnet: !secret ip_subnet
    dns1: !secret ip_dns1
    dns2: !secret ip_dns2
    
# Comment out and/or remove MQTT if you want to use the ESPHome API    
mqtt:
  broker: !secret mqtt_broker
  username: !secret mqtt_username
  password: !secret mqtt_password
  
# Enable logging
logger:

ota:
  safe_mode: True
  password: !secret esphome_otapass

font:
  - file: "fonts/Comic Sans MS.ttf"
    id: comic
    size: 23
  - file: "fonts/Comic Sans MS.ttf"
    id: comic1
    size: 16
  - file: "fonts/Comic Sans MS.ttf"
    id: comic2
    size: 31
  - file: "fonts/OpenSans-Semibold.ttf"
    id: opensans
    size: 11
    
time:
  - platform: sntp
    id: time1
    timezone: Asia/Kolkata
      
sensor:
  - platform: dht
    pin:
      number: GPIO5 # D1
      mode: INPUT_PULLUP
    model: AM2302
    temperature:
      name: "KBedroom Temperature"
      id: temperature
      unit_of_measurement: "°C"
      filters:
        filter_out: nan
    humidity:
      name: "KBedroom Humidity"
      id: humidity
      unit_of_measurement: "%"
      filters:
        filter_out: nan
    update_interval: 60s
  - platform: adc
    pin: A0
    name: "KBedroom Brightness"
    id: light_sensor
    update_interval: 60s
    unit_of_measurement: lux
    filters:
      - lambda: |-
          return (x / 10000.0) * 2000000.0;
      - filter_out: nan

  - platform: wifi_signal
    name: "KBedroom WiFi Signal"
    update_interval: 60s
    id: signal

binary_sensor:
  - platform: gpio
    pin: D5
    name: "KBedroom Motion"
    id: motion_sensor
    device_class: motion

i2c:
  sda: D6
  scl: D7
  scan: False

display:
  - platform: ssd1306_i2c
    model: "SSD1306 128x64"
    reset_pin: D0
    address: 0x3C
    id: my_display
    pages:
      - id: page1
        lambda: |-
          it.print(0, 0, id(opensans), "AGKP_KBedroom");
          it.printf(102, 5, id(opensans), "%.1f", id(signal).state);
          it.printf(5, 32, id(comic), "%.1f°C", id(temperature).state);
          it.printf(80, 35, id(comic1), "%.1f%%", id(humidity).state);
          it.printf(0, 60, id(opensans), "Motion: %s", id(motion_sensor).state ? "ON" : "OFF");
          it.printf(85, 60, id(opensans), "%.1flux", id(light_sensor).state);
      - id: page2
        lambda: |-
          it.strftime(0, 0, id(opensans), "%A %d-%b-%Y", id(time1).now());
          it.strftime(5, 10, id(comic2), "%l:%M%P", id(time1).now());
          it.printf(0, 60, id(opensans), "Motion: %s", id(motion_sensor).state ? "ON" : "OFF");
          it.printf(85, 60, id(opensans), "%.1flux", id(light_sensor).state);

interval:
  - interval: 5s
    then:
      - display.page.show_next: my_display
      - component.update: my_display
          
output:
  - platform: esp8266_pwm
    pin: D2
    id: redgpio
  - platform: esp8266_pwm
    pin: D3
    id: greengpio
  - platform: esp8266_pwm
    pin: D4
    id: bluegpio

light:
  - platform: rgb
    name: "KBedroom Led"
    red: redgpio
    green: greengpio
    blue: bluegpio
    effects:
      - random:
          name: "My Slow Random Effect"
          transition_length: 30s
          update_interval: 30s
      - random:
          name: "My Fast Random Effect"
          transition_length: 4s
          update_interval: 5s
      - strobe:
      - strobe:
          name: Strobe Effect With Red
          colors:
            - state: True
              brightness: 100%
              red: 100%
              green: 0%
              blue: 0%
              duration: 500ms
            - state: False
              duration: 250ms
            - state: True
              brightness: 100%
              red: 100%
              green: 0%
              blue: 0%
              duration: 500ms
      - strobe:
          name: Strobe Effect With Green
          colors:
            - state: True
              brightness: 100%
              red: 0%
              green: 100%
              blue: 0%
              duration: 500ms
            - state: False
              duration: 250ms
            - state: True
              brightness: 100%
              red: 0%
              green: 100%
              blue: 0%
              duration: 500ms
      - strobe:
          name: Police All
          colors:
            - state: True
              brightness: 100%
              red: 100%
              green: 0%
              blue: 0%
              duration: 500ms
            - state: False
              duration: 250ms
            - state: True
              brightness: 100%
              red: 0%
              green: 0%
              blue: 100%
              duration: 500ms
      - strobe:
          name: Water Motor
          colors:
            - state: True
              brightness: 100%
              red: 0%
              green: 0%
              blue: 100%
              duration: 200ms
            - state: False
              duration: 200ms
      - strobe:
          name: Ather Grid
          colors:
            - state: True
              brightness: 25%
              red: 0%
              green: 100%
              blue: 0%
              duration: 200ms
            - state: False
              duration: 200ms
            - state: True
              brightness: 15%
              red: 0%
              green: 100%
              blue: 0%
              duration: 200ms
      - strobe:
          name: Strobe Effect With Custom Values
          colors:
            - state: True
              brightness: 100%
              red: 100%
              green: 90%
              blue: 0%
              duration: 500ms
            - state: False
              duration: 250ms
            - state: True
              brightness: 100%
              red: 0%
              green: 100%
              blue: 0%
              duration: 500ms
      - flicker:
      - flicker:
          name: Flicker Effect With Custom Values
          alpha: 95%
          intensity: 1.5%

```

**Logs (if applicable):**

```
Archiving /data/kbedroom/.pioenvs/kbedroom/libFrameworkArduino.a
Linking /data/kbedroom/.pioenvs/kbedroom/firmware.elf
Retrieving maximum program size /data/kbedroom/.pioenvs/kbedroom/firmware.elf
Checking size /data/kbedroom/.pioenvs/kbedroom/firmware.elf

RAM:   [====      ]  42.0% (used 34376 bytes from 81920 bytes)
Flash: [====      ]  43.5% (used 454588 bytes from 1044464 bytes)
Creating BIN file "/data/kbedroom/.pioenvs/kbedroom/firmware.bin" using "/root/.platformio/packages/framework-arduinoespressif8266/bootloaders/eboot/eboot.elf" and "/data/kbedroom/.pioenvs/kbedroom/firmware.elf"
======================== [SUCCESS] Took 104.80 seconds ========================
INFO Successfully compiled program.
INFO Running:  esptool.py --before default_reset --after hard_reset --baud 460800 --chip esp8266 --port /dev/ttyUSB0 write_flash 0x0 /data/kbedroom/.pioenvs/kbedroom/firmware.bin
esptool.py v2.8
Serial port /dev/ttyUSB0
Connecting....
Chip is ESP8266EX
Features: WiFi
Crystal is 26MHz
MAC: 40:f5:20:33:a7:18
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Compressed 458736 bytes to 318431...
Wrote 458736 bytes (318431 compressed) at 0x00000000 in 7.3 seconds (effective 501.1 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
INFO Successfully uploaded program.
INFO Starting log output from /dev/ttyUSB0 with baud rate 115200
[21:23:05]\xfbo\xe0\x83l\x8c\xdcbg\xe3|\xecd\x87\x8fc\x8c\xfbog\xe7l\x8c\x8fl`\xd8\033\033gnl`n{\x87\x93\x9bo#d`{ۓncl`\x9cc\xc7\xdbl{\xdb`\xfc\x83n\x9c[I][logger:166]: Log initialized
[21:23:05][C][ota:366]: There have been 0 suspected unsuccessful boot attempts.
[21:23:05][I][app:029]: Running through setup()...
[21:23:05][D][binary_sensor:034]: 'KBedroom Motion': Sending initial state OFF
[21:23:05][C][esp8266_pwm:017]: Setting up ESP8266 PWM Output...
[21:23:05][C][esp8266_pwm:017]: Setting up ESP8266 PWM Output...
[21:23:05][C][esp8266_pwm:017]: Setting up ESP8266 PWM Output...
[21:23:05][C][light:097]: Setting up light 'KBedroom Led'...
[21:23:05][D][light:265]: 'KBedroom Led' Setting:
[21:23:05][D][light:278]:   Brightness: 100%
[21:23:05][D][light:287]:   Red=100%, Green=100%, Blue=100%
[21:23:05][C][dht:011]: Setting up DHT...
[21:23:05][C][adc:018]: Setting up ADC 'KBedroom Brightness'...
[21:23:05][C][ssd1306_i2c:010]: Setting up I2C SSD1306...
[21:23:05][W][i2c:070]: Received NACK on transmit of address 0x3C
[21:23:05][E][component:092]: Component was marked as failed.
[21:23:06][C][wifi:033]: Setting up WiFi...
[21:23:06][I][wifi:194]: WiFi Connecting to 'WIFI'...
[21:23:06][D][adc:056]: 'KBedroom Brightness': Got voltage=0.00V
[21:23:06][D][sensor:092]: 'KBedroom Brightness': Sending state 0.58594 lux with 2 decimals of accuracy
[21:23:06][D][dht:048]: Got Temperature=29.4°C Humidity=43.0%
[21:23:06][D][sensor:092]: 'KBedroom Temperature': Sending state 29.40000 °C with 1 decimals of accuracy
[21:23:06][D][sensor:092]: 'KBedroom Humidity': Sending state 43.00000 % with 0 decimals of accuracy
[21:23:06]
[21:23:06]--------------- CUT HERE FOR EXCEPTION DECODER ---------------
[21:23:06]
[21:23:06]Exception (29):
WARNING Exception type: Access to invalid address: STORE (wild pointer?)
[21:23:06]epc1=0x4020a092 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000
INFO Need to fetch platformio IDE-data, please stand by
INFO Running:  platformio run -d /config/esphome/kbedroom -t idedata
WARNING Decoded 0x4020a092: esphome::ssd1306_base::SSD1306::fill(esphome::Color)
[21:23:11]
[21:23:11]>>>stack>>>
WARNING Found stack trace! Trying to decode it
[21:23:11]
[21:23:11]ctx: cont
[21:23:11]sp: 3ffff450 end: 3fffffc0 offset: 0190
[21:23:11]3ffff5e0:  3fff4248 3fff3874 00000000 4020a09d  
WARNING Decoded 0x4020a09d: esphome::ssd1306_base::SSD1306::fill(esphome::Color)
[21:23:11]3ffff5f0:  3fff48a4 3fff48d4 3fff3874 4020a198  
WARNING Decoded 0x4020a198: non-virtual thunk to esphome::ssd1306_base::SSD1306::fill(esphome::Color)
[21:23:11]3ffff600:  000003e8 422c0000 3fff427c 40201a95  
WARNING Decoded 0x40201a95: esphome::display::DisplayBuffer::clear()
[21:23:11]3ffff610:  00003c58 0000078b 0000078b 4020206c  
WARNING Decoded 0x4020206c: esphome::display::DisplayBuffer::do_update_()
[21:23:11]3ffff620:  3fff3884 3fff48d4 3ffff660 3fff1834  
[21:23:11]3ffff630:  00000002 3fff182c 00000020 40209e75  
WARNING Decoded 0x40209e75: esphome::ssd1306_base::SSD1306::update()
[21:23:11]3ffff640:  60000000 403d6666 3fff3fe4 40222b66  
WARNING Decoded 0x40222b66: esphome::UpdateComponentAction<>::play()
[21:23:12]3ffff650:  00000008 3fff48d4 3ffef9f8 40222be0  
WARNING Decoded 0x40222be0: esphome::Action<>::play_complex()
[21:23:12]3ffff660:  00000000 3fff182c 3fff3e54 40222bf4  
WARNING Decoded 0x40222bf4: esphome::Action<>::play_complex()
[21:23:12]3ffff670:  422c0000 41eb3333 3ffef9ec 40222c14  
WARNING Decoded 0x40222c14: esphome::interval::IntervalTrigger::update()
[21:23:12]3ffff680:  3fff4814 fffee8d1 3ffef9ec 40222a20  
WARNING Decoded 0x40222a20: std::_Function_handler<void (), esphome::PollingComponent::call_setup()::{lambda()#1}>::_M_invoke(std::_Any_data const&) at component.cpp
[21:23:12]3ffff690:  3fff4814 3fff48d4 3ffef9ec 40100828  
WARNING Decoded 0x40100828: esphome::Scheduler::call()
[21:23:12]3ffff6a0:  3fff3c34 3fff07ec 3fff3c1c 3fff07ec  
[21:23:12]3ffff6b0:  00000000 0000000b 00000020 00000000  
[21:23:12]3ffff6c0:  000000a6 3fff48e4 0000ea60 4020d581  
WARNING Decoded 0x4020d581: std::pair<esphome::Component**, int> std::get_temporary_buffer<esphome::Component*>(int)
[21:23:12]3ffff6d0:  3fff07ec 0000000c 3ffef9ec 00000100  
[21:23:12]3ffff6e0:  3fff124c 0000000b 3ffef9ec 4020e522  
WARNING Decoded 0x4020e522: esphome::Application::setup()
[21:23:12]3ffff6f0:  3fff07ec 0000000c 00000020 40101942  
WARNING Decoded 0x40101942: malloc
[21:23:12]3ffff700:  00000100 3fff3c04 3fff3c34 4020d8c8  
WARNING Decoded 0x4020d8c8: esphome::Application::register_component_(esphome::Component*)
[21:23:12]3ffff710:  3fff467c 3ffffe90 00000010 3ffff740  
[21:23:12]3ffff720:  3fff46dc 3ffef9ec 3ffffe48 40214527  
WARNING Decoded 0x40214527: setup
[21:23:12]3ffff730:  3f800000 3f800000 feefeffe feefeffe  
[21:23:12]3ffff740:  3fff3cdc 3fff3e94 3fff3f2c 3fff4024  
[21:23:12]3ffff750:  3fff411c 3fff42e4 3fff432c 3fff446c  
[21:23:12]3ffff760:  3fff4564 3fff458c 3fff45dc 3f800000  
[21:23:12]3ffff770:  3f800000 3f800000 3f800000 000000fa  
[21:23:12]3ffff780:  3f800000 3f800000 00000000 3f800000  
[21:23:12]3ffff790:  00000000 3f800000 3f800000 000001f4  
[21:23:12]3ffff7a0:  3ffe970d 402600f8 00000000 00000004  
WARNING Decoded 0x402600f8: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff7b0:  00000004 0000000a 3ffe95dc 40260102  
WARNING Decoded 0x40260102: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff7c0:  00000000 00000004 00000004 0000000a  
[21:23:12]3ffff7d0:  3ffe970f 4026010c 00000000 00000004  
WARNING Decoded 0x4026010c: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff7e0:  00000006 00000008 3ffe9711 40260114  
WARNING Decoded 0x40260114: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff7f0:  00000000 0000000b 00000003 00000003  
[21:23:12]3ffff800:  3ffe9b7f 40260117 00000000 00000009  
WARNING Decoded 0x40260117: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff810:  00000004 00000003 3ffe95d1 4026011a  
WARNING Decoded 0x4026011a: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff820:  00000000 0000000a 00000003 00000002  
[21:23:12]3ffff830:  3ffe9b82 4026011c 00000000 00000004  
WARNING Decoded 0x4026011c: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff840:  00000006 00000008 3ffe96e3 40260124  
WARNING Decoded 0x40260124: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff850:  00000000 00000004 00000006 00000008  
[21:23:12]3ffff860:  3ffe9637 4026012c 00000000 00000004  
WARNING Decoded 0x4026012c: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff870:  00000006 00000008 3ffe953f 40260134  
WARNING Decoded 0x40260134: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff880:  00000000 00000004 00000006 00000008  
[21:23:12]3ffff890:  3ffe8e07 4026013c 00000000 00000004  
WARNING Decoded 0x4026013c: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff8a0:  00000007 00000008 3ffe9713 40260144  
WARNING Decoded 0x40260144: sleep_reset_analog_rtcreg_8266
[21:23:12]3ffff8b0:  00000000 00000004 00000006 00000008  
[21:23:12]3ffff8c0:  3ffe9585 4026014c 00000000 00000004  
WARNING Decoded 0x4026014c: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff8d0:  00000006 00000008 3ffe96d5 40260154  
WARNING Decoded 0x40260154: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff8e0:  00000000 00000004 00000006 00000008  
[21:23:13]3ffff8f0:  3ffe8e21 4026015c 00000000 00000004  
WARNING Decoded 0x4026015c: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff900:  00000006 00000008 3ffe9715 40260164  
WARNING Decoded 0x40260164: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff910:  00000000 00000004 00000006 00000008  
[21:23:13]3ffff920:  3ffe9717 4026016c 00000000 00000006  
WARNING Decoded 0x4026016c: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff930:  00000003 00000006 3ffe9148 40260172  
WARNING Decoded 0x40260172: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff940:  00000000 00000004 00000008 00000008  
[21:23:13]3ffff950:  3ffe97a3 4026017a 00000000 00000004  
WARNING Decoded 0x4026017a: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff960:  00000007 00000008 3ffe9652 40260182  
WARNING Decoded 0x40260182: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff970:  00000000 00000004 00000007 00000008  
[21:23:13]3ffff980:  3ffe88a0 4026018a 00000000 00000004  
WARNING Decoded 0x4026018a: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff990:  00000008 00000008 3ffe99ef 40260192  
WARNING Decoded 0x40260192: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff9a0:  00000000 00000004 00000006 00000008  
[21:23:13]3ffff9b0:  3ffe8891 4026019a 00000000 00000004  
WARNING Decoded 0x4026019a: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff9c0:  00000006 00000008 3ffe9b6a 402601a2  
WARNING Decoded 0x402601a2: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff9d0:  00000000 00000004 00000008 00000008  
[21:23:13]3ffff9e0:  3ffe9719 402601aa 00000000 00000004  
WARNING Decoded 0x402601aa: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffff9f0:  00000008 00000008 3ffe8ca7 402601b2  
WARNING Decoded 0x402601b2: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa00:  00000000 00000004 00000003 00000008  
[21:23:13]3ffffa10:  3ffe971b 402601ba ffffffff 00000004  
WARNING Decoded 0x402601ba: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa20:  00000005 0000000a 3ffe9166 402601c4  
WARNING Decoded 0x402601c4: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa30:  00000000 00000004 00000008 00000008  
[21:23:13]3ffffa40:  3ffe951e 402601cc 00000000 00000004  
WARNING Decoded 0x402601cc: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa50:  00000006 00000008 3ffe9af0 402601d4  
WARNING Decoded 0x402601d4: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa60:  00000000 00000004 0000000a 00000008  
[21:23:13]3ffffa70:  3ffe9990 402601e4 00000000 00000004  
WARNING Decoded 0x402601e4: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa80:  00000009 00000008 3ffe89fe 402601f4  
WARNING Decoded 0x402601f4: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffa90:  00000000 00000004 00000009 00000008  
[21:23:13]3ffffaa0:  3ffe9680 40260204 00000000 00000004  
WARNING Decoded 0x40260204: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffab0:  00000007 00000008 3ffe971d 4026020c  
WARNING Decoded 0x4026020c: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffac0:  00000000 00000004 00000009 0000000a  
[21:23:13]3ffffad0:  3ffe89f4 40260220 00000000 00000004  
WARNING Decoded 0x40260220: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffae0:  00000007 00000008 3ffe9acf 40260228  
WARNING Decoded 0x40260228: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffaf0:  00000000 00000004 00000006 00000008  
[21:23:13]3ffffb00:  3ffe9b1e 40260230 00000000 00000004  
WARNING Decoded 0x40260230: sleep_reset_analog_rtcreg_8266
[21:23:13]3ffffb10:  00000007 00000008 3ffe971f 40260238  
WARNING Decoded 0x40260238: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffb20:  00000000 00000004 00000008 00000008  
[21:23:14]3ffffb30:  3ffe8a1d 40260240 00000000 00000004  
WARNING Decoded 0x40260240: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffb40:  00000007 00000008 3ffe8a1a 40260248  
WARNING Decoded 0x40260248: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffb50:  00000000 00000004 0000000b 00000008  
[21:23:14]3ffffb60:  3ffe8ef8 40260258 00000000 00000004  
WARNING Decoded 0x40260258: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffb70:  00000007 00000008 3ffe9aed 40260260  
WARNING Decoded 0x40260260: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffb80:  00000000 00000004 00000007 00000008  
[21:23:14]3ffffb90:  3ffe8e5e 40260268 00000000 00000004  
WARNING Decoded 0x40260268: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffba0:  00000006 00000008 3ffe961d 40260270  
WARNING Decoded 0x40260270: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffbb0:  ffffffff 0000000c 00000007 00000002  
[21:23:14]3ffffbc0:  3ffe8cd5 40260272 00000000 00000006  
WARNING Decoded 0x40260272: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffbd0:  00000006 00000006 3ffe99f7 40260278  
WARNING Decoded 0x40260278: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffbe0:  00000000 00000004 00000007 00000008  
[21:23:14]3ffffbf0:  3ffe8db1 40260280 00000000 00000006  
WARNING Decoded 0x40260280: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc00:  00000006 00000006 3ffe99c3 40260286  
WARNING Decoded 0x40260286: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc10:  00000000 00000004 00000007 00000008  
[21:23:14]3ffffc20:  3ffe8af1 4026028e 00000000 00000006  
WARNING Decoded 0x4026028e: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc30:  00000006 00000006 3ffe95c7 40260294  
WARNING Decoded 0x40260294: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc40:  00000000 00000004 00000005 00000008  
[21:23:14]3ffffc50:  3ffe8be0 4026029c 00000000 00000006  
WARNING Decoded 0x4026029c: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc60:  00000006 00000009 3ffe9a36 402602a5  
WARNING Decoded 0x402602a5: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc70:  00000000 00000004 00000007 00000008  
[21:23:14]3ffffc80:  3ffe97ac 402602ad 00000000 00000004  
WARNING Decoded 0x402602ad: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffc90:  00000003 00000008 3ffe9721 402602b5  
WARNING Decoded 0x402602b5: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffca0:  ffffffff 00000004 00000005 0000000b  
[21:23:14]3ffffcb0:  3ffe9bf7 402602c0 00000000 00000004  
WARNING Decoded 0x402602c0: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffcc0:  00000007 00000008 3ffe8e65 402602c8  
WARNING Decoded 0x402602c8: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffcd0:  00000000 00000004 00000003 00000008  
[21:23:14]3ffffce0:  3ffe8e9c 402602d0 00000000 00000006  
WARNING Decoded 0x402602d0: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffcf0:  0000000b 00000006 3ffe9b2d 402602dc  
WARNING Decoded 0x402602dc: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffd00:  00000000 00000006 00000007 00000006  
[21:23:14]3ffffd10:  3ffe9723 402602e2 00000000 00000006  
WARNING Decoded 0x402602e2: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffd20:  00000007 00000006 3ffe9b1a 402602e8  
WARNING Decoded 0x402602e8: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffd30:  00000000 00000006 00000007 00000009  
[21:23:14]3ffffd40:  3ffe9725 402602f1 00000000 00000006  
WARNING Decoded 0x402602f1: sleep_reset_analog_rtcreg_8266
[21:23:14]3ffffd50:  00000007 00000009 3ffe99fb 402602fa  
WARNING Decoded 0x402602fa: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffd60:  00000000 00000006 00000005 00000006  
[21:23:15]3ffffd70:  3ffe9664 40260300 00000000 00000006  
WARNING Decoded 0x40260300: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffd80:  00000005 00000006 3ffe9a8d 40260306  
WARNING Decoded 0x40260306: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffd90:  00000000 00000005 00000005 00000007  
[21:23:15]3ffffda0:  3ffe8f23 4026030d 00000000 00000006  
WARNING Decoded 0x4026030d: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffdb0:  00000007 00000006 3ffe9a1b 40260313  
WARNING Decoded 0x40260313: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffdc0:  00000000 00000006 00000006 00000006  
[21:23:15]3ffffdd0:  3ffe9988 40260319 00000000 00000006  
WARNING Decoded 0x40260319: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffde0:  00000009 00000006 3ffe959e 40260325  
WARNING Decoded 0x40260325: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffdf0:  00000000 00000006 00000006 00000006  
[21:23:15]3ffffe00:  3ffe9ad8 4026032b 00000000 00000006  
WARNING Decoded 0x4026032b: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffe10:  00000006 00000009 3ffe9727 40260334  
WARNING Decoded 0x40260334: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffe20:  00000000 00000006 00000005 00000006  
[21:23:15]3ffffe30:  3ffe9729 4026033a 00000000 00000004  
WARNING Decoded 0x4026033a: sleep_reset_analog_rtcreg_8266
[21:23:15]3ffffe40:  00000005 00000008 3fff1370 00000000  
[21:23:15]3ffffe50:  00000000 3fff1390 00000000 00000000  
[21:23:15]3ffffe60:  00000001 40224098 963ca8c0 40224098  
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
[21:23:15]3ffffe70:  013ca8c0 40224098 00ffffff 40224098  
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
[21:23:15]3ffffe80:  013ca8c0 40224098 0100a8c0 00000000  
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
[21:23:15]3ffffe90:  3fff467c 3fff46a8 3fff46a8 013ca8c0  
[21:23:15]3ffffea0:  40224098 00ffffff 40224098 013ca8c0  
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
[21:23:15]3ffffeb0:  40224098 0100a8c0 3ffff740 00000003  
WARNING Decoded 0x40224098: std::__shared_count<(__gnu_cxx::_Lock_policy)0>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)0> const&)
[21:23:15]3ffffec0:  3ffff740 00000003 3ffffe90 00000001  
[21:23:15]3ffffed0:  3ffff740 00000002 3ffff740 00000003  
[21:23:15]3ffffee0:  3ffff740 00000003 3ffffe90 00000002  
[21:23:15]3ffffef0:  3ffff740 00000003 3ffffe90 00000001  
[21:23:15]3fffff00:  3ffff740 00000002 3ffff740 0000004b  
[21:23:15]3fffff10:  3ffff740 0000004b 3ffff740 0000004b  
[21:23:15]3fffff20:  3ffff740 0000004b 3fff4144 3fff3d14  
[21:23:15]3fffff30:  3fff382c 3fff377c 3fff370c 3fff3994  
[21:23:15]3fffff40:  feefeffe feefeffe feefeffe feefeffe  
[21:23:15]3fffff50:  3ffff740 3ffffc50 3ffffc68 3ffffc80  
[21:23:15]3fffff60:  3ffffc98 feefeffe 3fff467c 3ffefaa0  
[21:23:15]3fffff70:  3fff3874 00000003 3ffffe40 feefeffe  
[21:23:15]3fffff80:  feefeffe 3ffffa40 feefeffe feefeffe  
[21:23:15]3fffff90:  feefeffe feefeffe feefeffe 3ffefecc  
[21:23:15]3fffffa0:  3fffdad0 00000000 3ffefe8c 4021f3ac  
WARNING Decoded 0x4021f3ac: loop_wrapper() at core_esp8266_main.cpp
[21:23:15]3fffffb0:  feefeffe feefeffe 3ffe85c4 40100a45  
WARNING Decoded 0x40100a45: cont_wrapper
[21:23:15]<<<stack<<<
[21:23:15]
[21:23:52]--------------- CUT HERE FOR EXCEPTION DECODER ---------------
[21:23:52]
[21:23:52] ets Jan  8 2013,rst cause:2, boot mode:(3,6)
[21:23:52]
[21:23:52]load 0x4010f000, len 3584, room 16 
[21:23:52]tail 0
[21:23:52]chksum 0xb0
[21:23:52]csum 0xb0
[21:23:52]v2843a5ac
[21:23:52]~ld
[21:23:52][I][logger:166]: Log initialized
[21:23:52][C][ota:366]: There have been 10 suspected unsuccessful boot attempts.
[21:23:52][E][ota:371]: Boot loop detected. Proceeding to safe mode.
[21:23:52][I][app:029]: Running through setup()...
[21:23:52][C][wifi:033]: Setting up WiFi...
[21:23:52][I][wifi:194]: WiFi Connecting to 'WIFI'...
[21:23:52][W][wifi_esp8266:444]: Event: Disconnected ssid='WIFI' bssid=[redacted] reason='Handshake Failed'
[21:23:52][W][wifi:485]: Error while connecting to network.
[21:23:52][W][wifi:522]: Restarting WiFi adapter...
[21:23:52][I][wifi:194]: WiFi Connecting to 'WIFI'...
[21:23:52][I][wifi:457]: WiFi Connected!
[21:23:52][C][wifi:303]:   SSID: [redacted]
[21:23:52][C][wifi:304]:   IP Address: 192.168.0.23
[21:23:52][C][wifi:306]:   BSSID: [redacted]
[21:23:52][C][wifi:307]:   Hostname: 'kbedroom'
[21:23:52][C][wifi:311]:   Signal strength: -39 dB ▂▄▆█
[21:23:52][C][wifi:315]:   Channel: 11
[21:23:52][C][wifi:316]:   Subnet: 255.255.255.0
[21:23:52][C][wifi:317]:   Gateway: 192.168.0.1
[21:23:52][C][wifi:318]:   DNS1: 192.168.0.1
[21:23:52][C][wifi:319]:   DNS2: 192.168.0.1
[21:23:52][C][ota:029]: Over-The-Air Updates:
[21:23:52][C][ota:030]:   Address: 192.168.0.23:8266
[21:23:52][C][ota:032]:   Using Password.
[21:23:52][W][ota:036]: Last Boot was an unhandled reset, will proceed to safe mode in 0 restarts
[21:23:52][I][app:059]: setup() finished successfully!
[21:23:52][I][ota:381]: Waiting for OTA attempt.
[21:23:52][I][app:105]: ESPHome version 1.16.2 compiled on Feb 28 2021, 21:21:58
[21:23:52][C][wifi:443]: WiFi:
[21:23:52][C][wifi:303]:   SSID: [redacted]
[21:23:52][C][wifi:304]:   IP Address: 192.168.0.23
[21:23:52][C][wifi:306]:   BSSID: [redacted]
[21:23:52][C][wifi:307]:   Hostname: 'kbedroom'
[21:23:52][C][wifi:311]:   Signal strength: -39 dB ▂▄▆█
[21:23:52][C][wifi:315]:   Channel: 11
[21:23:52][C][wifi:316]:   Subnet: 255.255.255.0
[21:23:52][C][wifi:317]:   Gateway: 192.168.0.1
[21:23:52][C][wifi:318]:   DNS1: 192.168.0.1
[21:23:52][C][wifi:319]:   DNS2: 192.168.0.1
[21:23:52][C][logger:185]: Logger:
[21:23:52][C][logger:186]:   Level: DEBUG
[21:23:52][C][logger:187]:   Log Baud Rate: 115200
[21:23:52][C][logger:188]:   Hardware UART: UART0
[21:23:52][C][ota:029]: Over-The-Air Updates:
[21:23:52][C][ota:030]:   Address: 192.168.0.23:8266
[21:23:52][C][ota:032]:   Using Password.
[21:23:52][W][ota:036]: Last Boot was an unhandled reset, will proceed to safe mode in 0 restarts


```

**Additional information and things you've tried:**
1. Suspect the Display component SSD1306 (Received NACK on transmit of address 0x3C) - No go
2. Removed all sensors and flashed - No go
3. Flashed a new Wemos D1 as suspected old chip - No go
4. Changed Pin config - No go
5. Ran out of ideas. - Please help.

**Comment by mikosoft83** [comment]
I seem to be affected by this as well. It works for a while (a while = random time interval between 10 minutes and two days) and then just gets stuck at setup(). Tried two D1 minis, same hardware setup, same screen as above, same result. If I remove I2C seems to work just fine.

**Comment by ioios-io** [comment]
Same here, not sure which update it was but having the exact same issues as Mikosoft83. All 3 of us are using the ssd1306_i2c screen, I think that's the only thing in common.
```
i2c:
  sda: ${i2cData}
  scl: ${i2cClock}
  scan: true
  frequency: 200MHz
  id: bus_a

font:
  - file: "fonts/LiberationSans-Regular.ttf"
    id: title_font
    size: 12
    glyphs: ":/HTPS"

image:
  - file: "images/return.png"
    id: return_image
    resize: 48x48

display:
  - platform: ssd1306_i2c
    model: "SH1106 128x64"
    address: 0x3C
    update_interval: 0.2s
    id: i2cDisplay
    pages:
```

**Comment by wwebers** [comment]
Has anyone of you tried to raise the update intervall to 10s for the display? What I realized is that the communication with the display over I2C is quiet slow and very talkative. Which results the display driver sending a lot of data over the I2C bus to the display (even when not showing anything) and blocking the WiFi routines and sometimes it seem to prevent the ESP8266 watchdog timer running nicely.

When I raised the update_interval to 10s everything went nice. However, 10s is quiet a lot. Better solution would be to "shut up" that talkative display library...

**Comment by kk-sonata** [comment]
I wonder if there's not enough heap memory to allocate the buffer for the display. I'm working on a similar issue with the st7789 display. ESPhome should have more elegant error messages in these cases.

**Comment by glmnet** [comment]
Duplicate of #1628

**Comment by FreshJonte** [comment]
Having a similar issue with the ST7735 display on D1 mini pro. When "update_interval" triggers and the display updates, it loses network connectivity. Drops packets and makes OTA fail for example.

This is the log (VERBOSE):
[21:56:16][V][app:081]: A component took a long time in a loop() cycle (0.95 s).
[21:56:16][V][app:082]: Components should block for at most 20-30ms in loop().

**Comment by wwebers** [comment]
> Duplicate of #1628

Don't think its the same issue. The one you refer to is about a boot loop. This issue is not a boot loop as the board perfectly boots, but WiFi is dead because of too much traffic on the I2C bus produced by the display library.

**Comment by PrathikGopal** [comment]
So, v1.17.0 doesn't fix it. Hopefully someone can resolve this issue.

**Comment by mikosoft83** [comment]
Same with me, I updated yesterday, reflashed my device but it got stuck just like before.

**Comment by PrathikGopal** [comment]
Boot is ok on v1.17.1, but the device is not found

```
[23:57:03][I][i2c:033]: Scanning i2c bus for active devices...
[23:57:03][I][i2c:049]: Found no i2c devices!
[23:57:04][C][ssd1306_i2c:023]: I2C SSD1306
[23:57:04][C][ssd1306_i2c:023]:   Rotations: 0 °
[23:57:04][C][ssd1306_i2c:023]:   Dimensions: 128px x 64px
[23:57:04][C][ssd1306_i2c:024]:   Address: 0x3C
[23:57:04][C][ssd1306_i2c:025]:   Model: SSD1306 128x64
[23:57:04][C][ssd1306_i2c:026]:   Reset Pin: GPIO16 (Mode: OUTPUT)
[23:57:04][C][ssd1306_i2c:028]:   Update Interval: 1.0s
[23:57:04][E][ssd1306_i2c:031]: Communication with SSD1306 failed!
```

**Comment by PrathikGopal** [comment]
Well what do you know, it was a the damn jumper wires.... The whole bunch of them. I used a wire from a newer lot and it works. 
Thankfully the errors prompted me to relook on the connected cables.
